doc/source/reference/demeshening.rst

```diff
@@ -0,0 +1,146 @@
+     which particles are contained -- assign each particle to an integer value
+     according to the space-filling curve and the coarse index order.  Set the
+     "bit" in an array of boolean values that each of these integers correspond
+     to.  Note that this is almost certainly *reductive* -- there will be fewer
+     bits set than there are particles, which is *by design*.
+  2. Once all chunks or files have been assigned an array of bits that
+     correspond to the places where, according to the coarse indexing scheme,
+     they have particles, identify all those "bits" that have been set by more
+     than one chunk.  All of these bits correspond to locations where more than
+     one file contains particles -- so if you want to select something from
+     this region, you'd need to read more than one file.
+  3. For each "collision" location, apply a *second-order* index, to identify
+     which sub-regions are touched by more than one file.
+
+At the end of this process, each file will be associated with a single "coarse"
+index (which covers the entire domain of the data), as well as a set of
+"collision" locations, and in each "collision" location a set of bitarrays that
+correspond to that subregion.
+
+When reading data, yt will identify which "coarse" index regions are necessary
+to read.  If any of those coarse index regions are covered by more than one
+file, it will examine the "fine" index for those regions and see if it is able
+to subset more efficiently.  Because all of these operations can be done with
+logical operations, this considerably reduces the amount of data that needs to
+be read from disk before expensive selection operations are conducted.
+
+For those situations that involve particles with regions of influence -- such
+as smoothed particle hydrodynamics, where particles have associated smoothing
+lenghts -- these are taken into account when conducting the indexing system.
+
+Efficiency of Index Orders
+--------------------------
+
+What this can lead to, however, is situations where (particularly at the edges
+of regions populated by SPH particles) the indexing system identifies
+collisions, but the relatively small number of particles and correspondingly
+large "smoothing lengths" result in a large number of "fine" index values that
+need to be set.
+
+Counterintuitively, this actually means that occasionally the "fine" indexing
+process can take an inordinately long amount of time for *small* datasets,
+rather than large datasets.
+
+In these situations, it is typically sufficient to set the "fine" index order
+to be much lower than its default value.  For instance, setting the
+``index_order`` to (5, 3) means that the full domain will be subdivided into 32
+bins in each dimension, and any "collision" zones will be further subdivided
+into 8 bins in each dimension (corresponding to an effective 256 bins across
+the full domain).
+
+If you are experiencing very long index times, this may be a productive
```

Comment:
How long is "very" ? From reading this section I feel like the first thing I'd want to do before processing a lot of data is to find the optimal values in `index_order` for my application, but I don't know really how much I should expect to gain from it or loose if I can't afford this optimisation step.

Reply:
Added some discussion!